connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/ClusterConfigState.java

```diff
@@ -86,6 +114,20 @@ public long offset() {
     }
 
     /**
+     * Get all task configs for a connector.
+     * @param connector name of the connector
+     * @return a map from the task id to its configuration
+     */
+    public Map<ConnectorTaskId, Map<String, String>> allTaskConfigs(String connector) {
```

Comment:
Hmm, we may end up wanting to reorganize some collections to be like `Map<Connector, Map<TaskId, Map<String, String>>>` instead. Probably not an issue since given the # of connectors/tasks and the infrequent need to do this collection, but full scans like this make me unhappy.